docs/adr/012_decouple_from_edge.md

```diff
@@ -0,0 +1,147 @@
+  - Closing DeviceClient connections for avoiding "ping-pong" as described in [ADR 010](./010_lns_affinity.md)
+  - Sending Cloud to Device messages for Class-C devices
+
+Except for ModuleConnectionHost, the other difference is concerning the ability to "proxy" leaf-device messages through Edge Hub. By default the "ENABLE_GATEWAY" environment variable is set to true, we need to make sure that **ENABLE_GATEWAY can't be set to true when not running as Edge module**
+
+### Differentiate Edge from Cloud deployments
+
+As of v2.1.0, the LNS code is already able to understand whether it is running as IoT Edge Module or not. The availability of the "IOTEDGE_APIVERSION" environment variable is checked. This variable is injected by the "Edge Agent" component at the moment of module creation and startup, therefore if the variable is not there we can safely assume that the LNS is not running in a Edge environment.
+
+When a leaf device message is sent and processed, the device twin gets updated with the "GatewayID", currently being set to the "IOTEDGE_DEVICEID" environment variable.
+
+This information might be used by the Azure Function:
+
+- for sending the LNS a command to close the device client connection for such leaf device
+- for sending a C2D message for a Class-C device through a "direct method invocation"
+
+We need to **conditionally set the "GatewayID" parameter depending on whether the LNS is running on Edge or not**
+
+The proposal is to:
+
+- Prefix the "GatewayID" with "standalone-" for non-Edge deployments
+- Use the "Hostname" environment variable as a default for GatewayID field
+
+### Managing configuration and its changes
+
+All the configuration is fetched at the start time from the environment variables.
+
+IoT Edge allows you to specify those configuration variables as part of a template for [automatic deployments for single devices or at scale](https://docs.microsoft.com/en-us/azure/iot-edge/module-deployment-monitoring?view=iotedge-2020-11)
+
+Depending on the target cloud service, you might have different ways to manage configuration in a centralized way.
+
+If using Azure Kubernetes Service, a configmap to be applied to the desired cluster might work for you.
+If using Azure App Service, you might just set the environment variables in the configuration section of the application.
+
+While these solutions are great for "static" configuration (i.e.: those values that require a restart for being changed), the LNS is coded in such a way that some configuration might be changed dynamically without having to restart the entire network server.
+
+The proposal is to **make use of "[Azure App Configuration](https://docs.microsoft.com/en-us/azure/azure-app-configuration/overview)"** to centrally manage application settings and keep [dynamic configuration using poll model](https://docs.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-aspnet-core?tabs=core5x)
+
+### Direct method invocation
```

Comment:
consider this a stretch - alternative is to have the configuration changes requiring restarts.

Reply:
I will make the easier way of doing it the preferred way and put the Azure App Configuration as an alternative